lib/SDIR/Translate/translateToSDFG.cpp

```diff
@@ -0,0 +1,1597 @@
+
+  argname.erase(0, 1); // remove %-sign
+  jemit.printKVPair("dst_connector", argname);
+
+  jemit.endObject();
+  return success();
+}
+
+LogicalResult translateCallToSDFG(sdir::CallOp &op, JsonEmitter &jemit) {
+  if (op.callsTasklet()) {
+    TaskletNode task = op.getTasklet();
+    for (unsigned i = 0; i < op.getNumOperands(); ++i) {
+      Value val = op.getOperand(i);
+      if (LoadOp load = dyn_cast<LoadOp>(val.getDefiningOp())) {
+        if (printLoadTaskletEdge(load, task, i, jemit).failed())
+          return failure();
+      } else if (sdir::CallOp call =
+                     dyn_cast<sdir::CallOp>(val.getDefiningOp())) {
+        TaskletNode taskSrc = call.getTasklet();
+        if (printTaskletTaskletEdge(taskSrc, task, i, jemit).failed())
+          return failure();
+      } else if (arith::ConstantOp con =
+                     dyn_cast<arith::ConstantOp>(val.getDefiningOp())) {
+        // TODO: Add constants
+      } else {
+        mlir::emitError(op.getLoc(), "Operands must be results of GetAccessOp, "
+                                     "LoadOp, TaskletNode or ConstantOp");
+        return failure();
+      }
+    }
+  } else {
+    // calls nested SDFG
+    SDFGNode sdfg = op.getSDFG();
+    for (unsigned i = 0; i < op.getNumOperands(); ++i) {
+      Value val = op.getOperand(i);
+      if (GetAccessOp acc = dyn_cast<GetAccessOp>(val.getDefiningOp())) {
+        if (printAccessSDFGEdge(acc, sdfg, i, jemit).failed())
+          return failure();
+      } else {
+        mlir::emitError(op.getLoc(),
+                        "Operands must be results of GetAccessOp, LoadOp, "
+                        "TaskletNode or ConstantOp");
+        return failure();
+      }
+    }
+  }
+
+  return success();
+}
+
+//===----------------------------------------------------------------------===//
+// LibCallOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult translateLibCallToSDFG(LibCallOp &op, JsonEmitter &jemit) {
+  return success();
+}
+
+//===----------------------------------------------------------------------===//
+// AllocSymbolOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult translateAllocSymbolToSDFG(AllocSymbolOp &op,
+                                         JsonEmitter &jemit) {
+  jemit.printKVPair(op.sym(), "int64");
+  return success();
+}
+
+//===----------------------------------------------------------------------===//
+// SymbolExprOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult translateSymbolExprToSDFG(SymOp &op, JsonEmitter &jemit) {
+  return success();
+}
+
+//===----------------------------------------------------------------------===//
+// Translate type
+//===----------------------------------------------------------------------===//
+
+LogicalResult translateTypeToSDFG(Type &t, Location loc, JsonEmitter &jemit,
```

Comment:
You should probably use this method when you lift to python or hardcode types. :)